audio_analysis.py

```diff
@@ -18,131 +19,83 @@ def get_wav_properties(wav_path):
     return fs, duration
 
 
-def audio_based_feature_extraction(input_file):
-
+def silence_features(segment_limits,dur):
+    """
+    Extract silence features based on audio segments
+    :param segment_limits: A list of start-end timestamps for audio segments
+    :param dur: The total duration of audio file
+    :return: A list of silence_features, number_of_pauses, total_speech
+    """
+    total_speech = 0.0
+    silence_durations = []
+    counter = 0
+    for k in segment_limits:
+        if counter == 0:
+            silence_durations.append(k[0])
+        elif counter == (len(segment_limits) - 1):
+            p = counter - 1
+            silence_durations.append(k[0] - segment_limits[p][1])
+            silence_durations.append(dur - k[1])
+        else:
+            p = counter - 1
+            silence_durations.append(k[0] - segment_limits[p][1])
+        word_speech = k[1] - k[0]
+        total_speech = total_speech + word_speech
+        counter = counter + 1
+    speech_ratio = total_speech / dur
+    speech_ratio = float("{:.2f}".format(speech_ratio))
+    number_of_pauses = len(silence_durations)
+    silence_durations = np.array(silence_durations)
+    std = np.std(silence_durations)
+    std = float("{:.2f}".format(std))
+    average_silence_dur = np.mean(silence_durations)
+    average_silence_dur = float("{:.2f}".format(average_silence_dur))
+    silence_seg_per_minute = float("{:.2f}".format(number_of_pauses /
+                                                   (dur / 60.0)))
+    word_rate_in_speech = len(segment_limits) / total_speech
```

Comment:
there's also a case of div zero here. I will fix in the next commit before merging @sofiaele